model/common/tests/grid_tests/test_geometry.py

```diff
@@ -0,0 +1,402 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+import functools
+
+import numpy as np
+import pytest
+
+from icon4py.model.common import dimension as dims
+from icon4py.model.common.decomposition import definitions
+from icon4py.model.common.grid import (
+    geometry as geometry,
+    geometry_attributes as attrs,
+    icon,
+    simple as simple,
+)
+from icon4py.model.common.grid.geometry import as_sparse_field
+from icon4py.model.common.test_utils import datatest_utils as dt_utils, helpers
+from icon4py.model.common.utils import gt4py_field_allocation as alloc
+
+from . import utils
+
+
+grid_geometries = {}
+
+
+def get_grid_geometry(backend, grid_file):
+    def construct_decomposition_info(grid: icon.IconGrid):
+        edge_indices = alloc.allocate_indices(dims.EdgeDim, grid)
+        owner_mask = np.ones((grid.num_edges,), dtype=bool)
+        decomposition_info = definitions.DecompositionInfo(klevels=grid.num_levels)
+        decomposition_info.with_dimension(dims.EdgeDim, edge_indices.ndarray, owner_mask)
+        return decomposition_info
+
+    def construct_grid_geometry(grid_file: str):
+        gm = utils.run_grid_manager(grid_file)
+        grid = gm.grid
+        decomposition_info = construct_decomposition_info(grid)
```

Comment:
since `construct_decomposition_info` is only used here, would it make sense to merge these two functions?